### DISCUSSION TITLE
CaloCluster drawing locations

### DISCUSSION THREAD

**Discussion by lawrenceleejr** [HSF/phoenix#225] [Q&A] [answered]
Hi! I was playing with Phoenix and injecting some objects via JSON. I'm seeing unexpected locations of my CaloCluster objects in eta/phi space. If I, for example, hand the same eta/phi direction to both a jet and a CaloCluster object, they're in very different locations. I suspect the issue is somewhere in the projection surface handling, probably on my side. 

The following screenshot shows a jet and a CaloCluster object with exactly the same eta/phi, and you can see the shift in drawing location. The JSON snippet is also below.

![Screen Shot 2021-03-03 at 5 39 03 PM](https://user-images.githubusercontent.com/7438278/109839396-5aa1af80-7c47-11eb-9c39-8ee44bec42d8.png)

```JSON
        "Jets": {
            "Jets": [
                {
                    "phi": -0.6605201959609985,
                    "eta": -0.3017900288105011,
                    "energy": 633767.0
                },
                {
                    "phi": -0.6605201959609985,
                    "eta": -0.30008089542388916,
                    "energy": 502159.03125
                }
            ]
        },
        "CaloClusters": {
            "JetConstituents": [
                {
                    "phi": -0.6605201959609985,
                    "eta": -0.30008089542388916,
                    "energy": 502159.03125
                }
            ]
        },
```

I see in the manual:

> Uniquely for clusters, you need to define the plane(s) on which to project the clusters as a property of the collection itself, using the following notation
> `'CYL':[30.2493,243.645,136.947,213.396,3.14159,2850]`

but I can't quite figure out 1) where exactly to add this to my collection in a way that makes it have any effect, and 2) what these numbers specifically correspond to. I'm hoping someone can provide some feedback on how to think about the positioning of the CaloCluster objects.

Thanks!
-Larry

**Comment by 9inpachi** [comment]
Hi @lawrenceleejr 

I am not a physics guy so I can't really answer you on the domain aspect. I think @EdwardMoyse will when he sees it.

But going through the code I don't see `CYL` (plane definition) being used anywhere. It most probably got lost at some point in our code switching or was never a part of it.

Even so, if you want to have a look at the object construction logic for clusters. The code is here: https://github.com/HSF/phoenix/blob/master/packages/phoenix-event-display/src/loaders/objects/phoenix-objects.ts#L200-L228

**Reply by EdwardMoyse** [reply]
Yes, I'm afraid that CYL did get lost. We need to add it back in, or make it a configurable parameter.

But this also just looks like a weird bug. I need to find some time to really debug it. Hopefully tomorrow! Thanks very much for reporting it @lawrenceleejr

**Reply by lawrenceleejr** [reply]
Thanks both! Was curious when I couldn't find `CYL` in the source. Looking at the function @9inpachi pointed to -- did a minus sign get lost in the theta calculation? 

https://github.com/HSF/phoenix/blob/master/packages/phoenix-event-display/src/loaders/objects/phoenix-objects.ts#L210

I'm not sure if produces the issue I'm seeing -- getting too late in the day to think through that...

**Reply by EdwardMoyse** [reply]
Possibly this is the issue ... @9inpachi, we could probably think of a unit test to give all objects the same parameters and check they have the same position/direction? Anyway, I'll dig into it tomorrow. And we should compare to a few 'known' events to look by eye for other issues.

**Comment by EdwardMoyse** [comment]
Hi all,

Sorry for the delay - I can confirm there is a bug with the calo positions (I made a test JSON and used labels to check). I think we can fix it easily but I would prefer clean all of this code up a bit, so it might have to wait until next week.

![image](https://user-images.githubusercontent.com/6764617/112712025-f633e200-8ecc-11eb-93f3-9ed466d18a95.png)

**Comment by EdwardMoyse** [comment, marked as answer]
Hi Larry - have a look at this MR where I think I fixed it #251

**Reply by lawrenceleejr** [reply]
Thanks so much! Unfortunately I don't have an active dev environment or anything, but would be happy to take a look and play around once there's a new tag pushed to CDN. (By the way -- I LOVE the CDN availability. I've been playing around with it. Thanks so much for that.)
-L

**Reply by 9inpachi** [reply]
> By the way -- I LOVE the CDN availability. I've been playing around with it. Thanks so much for that.

That's great Larry! You're welcome.
Just wanted to say that if you encounter any problem or limitation in CDN - let us know. ;)
